arch/arm/boot/dts/overlays/sunfounder-pironman5-overlay.dts

```diff
@@ -0,0 +1,52 @@
+/dts-v1/;
+/plugin/;
+
+/ {
+    compatible = "brcm,bcm2835";
+
+    fragment@0 {
+        target = <&i2c1>;
+        __overlay__ {
+            status = "okay";
+        };
+    };
+    fragment@1 {
+        target = <&spi0>;
+        __overlay__ {
+            status = "okay";
+        };
+    };
+    fragment@2 {
+        target-path = "/";
+        __overlay__ {
+            gpio_ir: ir-receiver@12 {
```

Comment:
The convention is to use hexadecimal "addresses" after the "@". The "12" here is 0x12, i.e. 18, which matches the gpio-ir overlay you've based this on. Because you've changed the GPIO to (decimal/denary) 12, you should also change the address to "@c". In that way it can coexist with a regular gpio-ir overlay on GPIO 18.